backend/analytics_server/app.py

```diff
@@ -36,5 +37,19 @@
 configure_db_with_app(app)
 initialize_database(app)
 
+def handle_exception(e):
```

Comment:
The handler is implemented well, but we need to let the flask know that this is the Error Handler it should be using for unhandled errors(Because flask returns with HTML Content when an unexpected/unhandled error occurs)
Check this: https://flask.palletsprojects.com/en/2.3.x/errorhandling/#registering

so adding something like below to `app` should suffice:
`
app.register_error_handler(Exception, handle_exception)`